libs/server/Resp/Objects/ListCommands.cs

```diff
@@ -679,8 +679,12 @@ private unsafe bool ListMove<TGarnetApi>(int count, byte* ptr, ref TGarnetApi st
                 if (!RespReadUtils.ReadPtrWithLengthHeader(ref param2.ptr, ref param2.length, ref ptr, recvBufferPtr + bytesRead))
                     return false;
 
-                var sourceDirection = param1.ReadOnlySpan.SequenceEqual("RIGHT"u8) ? OperationDirection.Right : OperationDirection.Left;
-                var destinationDirection = param2.ReadOnlySpan.SequenceEqual("RIGHT"u8) ? OperationDirection.Right : OperationDirection.Left;
+                OperationDirection sourceDirection = GetOperationDirection(param1);
+                OperationDirection destinationDirection = GetOperationDirection(param2);
+                if (sourceDirection == OperationDirection.Unknown || destinationDirection == OperationDirection.Unknown)
+                {
+                    return AbortWithErrorMessage(count, CmdStrings.RESP_ERR_GENERIC_SYNTAX_ERROR);
```

Comment:
Thanks for adding this fix! If you have a couple of minutes, would you please add a couple of test cases to check that the correct error is raised? Probably best to add in the CanUseLMoveGC method in RespListTest.cs
Appreciate it!

Reply:
Of course. I'm currently participating in a competition outside, I'll add it after the competition.

Reply:
Bonus points if you can also do a similar thing in ListInsert for BEFORE | AFTER string comparisons :)

Reply:
> Bonus points if you can also do a similar thing in ListInsert for BEFORE | AFTER string comparisons :) (No pressure, we can always add that later)

I have added test case. BEFORE | AFTER string comparisons I will finish it later :)

Reply:
Sounds good, appreciate it!